tests/commons/src/main/java/org/codice/ddf/admin/comp/graphql/GraphQlResponse.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.comp.graphql;
+
+import com.google.common.collect.ImmutableList;
+import com.jayway.restassured.response.ExtractableResponse;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import org.apache.commons.collections.CollectionUtils;
+import org.codice.ddf.admin.api.report.ErrorMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @param <T> the expected return type of the object in the json response identified by the
+ *     jsonResultPath parameter
+ */
+public class GraphQlResponse<T> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GraphQlResponse.class);
+
+  public static final String GRAPHQL_ERRORS = "errors";
+
+  private final List<ErrorMessage> errors;
+
+  private T result;
+
+  /**
+   * Creates a GraphQL response from an {@link ExtractableResponse}. If there are errors in the
+   * response, the jsonResultPath will be ignored and errors in the {@code GraphQlResponse} will be
+   * populated.
+   *
+   * @param response response of the GraphQL request
+   * @param jsonResultPath json path to the expected result of the GraphQL response
+   */
+  public GraphQlResponse(
+      @Nonnull final ExtractableResponse response, @Nonnull final String jsonResultPath) {
```

Comment:
Piggy backing off my last comment, I think we should store the response as a variable and only invoke `jsonPath` parameter through the `getResult` method